exporter/opentelemetry-exporter-jaeger/src/opentelemetry/exporter/jaeger/__init__.py

```diff
@@ -109,23 +109,26 @@ def __init__(
         service_name,
         agent_host_name=DEFAULT_AGENT_HOST_NAME,
         agent_port=DEFAULT_AGENT_PORT,
-        collector_host_name=None,
-        collector_port=None,
-        collector_endpoint=DEFAULT_COLLECTOR_ENDPOINT,
-        collector_protocol=DEFAULT_COLLECTOR_PROTOCOL,
+        collector_endpoint=None,
```

Comment:
Isn't the default supposed to be `http://localhost:14250`?

Reply:
I think in this case it shouldn't. Previously `collector_endpoint` was related to `/api/traces?format=jaeger.thrift` and it was just a part of complete endpoint address. If we set `collector_endpoint` default then it will be used over `agent` which was default implementation.

Reply:
Any reason why the specs would have that is the default value then? 

> If we set collector_endpoint default then it will be used over agent which was default implementation

I'm also not sure what you mean by the above statement.

Reply:
I just want to say that in previous implementation `collector` by default was not used and `collector_endpoint` was just an end part of complete URL. In current implementation there's no need to specify host,port and endpoint separately because they will be provided by one environment variable called `OTEL_EXPORTER_JAEGER_ENDPOINT` coming from otel specification.

Reply:
Correct, I am asking whether the `collector_endpoint` value should default to `http://localhost:14250` as defined by the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-environment-variables.md) if the value is not explicitly passed into the exporter AND the environment variable is not configured.

Reply:
Ahh, yes @lzchen, you're right :) I will add it. Thanks for heads up 👍

Reply:
According to our discussion on the Gitter. We've decided to do not set the `collector_endpoint` default because it is causing not entirely clear path of the used protocol to export spans. Separate issue and PR should be created for this purpose.

Reply:
Hmm, looking at [java's](https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java#L263) implementation, it looks like they only have the concept of the [collector](https://github.com/open-telemetry/opentelemetry-java/blob/master/exporters/jaeger/src/main/java/io/opentelemetry/exporters/jaeger/JaegerGrpcSpanExporter.java#L36) endpoint. This leads me to believe that we should still have the default value for `collector_endpoint`, and if there's anything that errors with the [creation](https://github.com/open-telemetry/opentelemetry-python/blob/master/exporter/opentelemetry-exporter-jaeger/src/opentelemetry/exporter/jaeger/__init__.py#L158) of the `Collector`, then we would use the [AgentClientUDP](https://github.com/open-telemetry/opentelemetry-python/blob/master/exporter/opentelemetry-exporter-jaeger/src/opentelemetry/exporter/jaeger/__init__.py#L172) instead.

Reply:
@codeboten @rydzykje 
Thoughts on this?

Reply:
Looking at the code, I don't know if there's any scenario where the `.constructor` property would return `None` if the default of `collector_endpoint` is always set. Based on the [jaeger docs](https://www.jaegertracing.io/docs/1.19/faq/#do-i-need-to-run-jaeger-agent), it would appear that only the agent configuration is usually set by default, and a collector endpoint would have to be explicitly set:

>If we want Jaeger client libraries to send trace data directly to collectors, we must provide them with a URL of the HTTP endpoint. It means that our applications require additional configuration containing this parameter, especially if we are running multiple Jaeger installations

I wonder if the spec should default to not having a collector endpoint. You're right @lzchen that other implementations only have the concept of an endpoint, [go](https://github.com/open-telemetry/opentelemetry-go/blob/master/exporters/trace/jaeger/env.go#L38) does the same thing.

Reply:
I would be ok to go ahead with this implementation and raise an issue with the spec about this.